### ISSUE TITLE
Defer calling canEnableFFmpegOptimizations on module load

### ISSUE THREAD

**Issue by JMTK** [discordjs/discord.js#9859]
### Which package is this bug report for?

@discord.js/voice

### Issue description

On app startup, I noticed that there were calls to `canEnableFFmpegOptimizations()`, despite not actually doing anything with voice for some testing I was doing. I do include discord.js/voice code elsewhere. It seems like in https://github.com/discordjs/voice/blob/309ac8596cac422cf22e51331869e011c720124c/src/audio/TransformerGraph.ts#L169, it is calling it just when the module is require/imported. I think this could be moved elsewhere somehow to only call it the first time the nodes are actually referenced somewhere, which should improve app startup time.

### Code sample

1. bot.js:

```js
import Discord from 'discord.js';
import * as DiscordVoice from '@discord.js/voice';
```

2. Run `node --prof bot.js`

3. Run `node --prof-process isolate...log > processed.txt`:
```
 [C++ entry points]:
   ticks    cpp   total   name
    138    1.3%    JS: ~spawnSync node:internal/child_process:1109:19
    138  100.0%      JS: ~spawnSync node:child_process:831:19
    138  100.0%        JS: ~getInfo /discordbot/node_modules/prism-media/src/core/FFmpeg.js:123:17
    138  100.0%          JS: ~canEnableFFmpegOptimizations /discordbot/node_modules/@discordjs/voice/dist/index.js:2234:38
    138  100.0%            JS: ~<anonymous> /discordbot/node_modules/@discordjs/voice/dist/index.js:1:1
```

### Versions

- discord.js: 14.13.0
- @discordjs/voice: 0.16.0

### Issue priority

Low (slightly annoying)

### Which partials do you have configured?

Not applicable

### Which gateway intents are you subscribing to?

Not applicable

**Comment by balogun14** [comment]
@JMTK can you explain this a bit more

**Comment by JMTK** [comment]
> @JMTK can you explain this a bit more

Sure!

I was recently profiling my bot to investigate sporadic CPU spikes. While I was doing this, unrelated to anything I was actually looking for, I noticed that every time I was profiling on bot startup without doing anything, I saw `spawnSync > ffmpeg > canEnableFFmpegOptimizations` in the call stack of the processed CPU profile from using [node.js CPU profiling](https://nodejs.org/en/docs/guides/simple-profiling). Upon further investigation, I found that it happens inside `TransformerGraph.ts`, right in the root of the file as soon as it is imported that function executes.|

![image](https://github.com/discordjs/discord.js/assets/877114/5725d278-9dcf-4cf1-bc39-238797b12dc5)

(IMO) Ideally someone who just importing the `voice` module shouldn't have to worry about importing it and having synchronously spawned processes to get metadata about FFmpeg without even using ffmpeg in that moment. It is negatively affecting the performance of starting my bot if I use `import` statements since they're all done as soon as the module loads rather than lazy loaded which would require code changes away from "best practices" of ES6.
